src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioErrorReportingService.ExceptionFormatting.cs

```diff
@@ -0,0 +1,210 @@
+                {
+                    continue;
+                }
+                if (firstFrame)
+                {
+                    firstFrame = false;
+                }
+                else
+                {
+                    stringBuilder.Append(Environment.NewLine);
+                }
+
+                stringBuilder.AppendFormat("   {0} ", ServicesVSResources.at);
+                var isAsync = FormatMethodName(stringBuilder, declaringType);
+                if (!isAsync)
+                {
+                    stringBuilder.Append(method.Name);
+                    var methodInfo = method as MethodInfo;
+                    if (methodInfo != null && methodInfo.IsGenericMethod)
+                    {
+                        FormatGenericArguments(stringBuilder, methodInfo.GetGenericArguments());
+                    }
+                }
+                else if (declaringType?.IsGenericType == true)
+                {
+                    FormatGenericArguments(stringBuilder, declaringType.GetGenericArguments());
+                }
+
+                stringBuilder.Append("(");
+                if (isAsync)
+                {
+                    stringBuilder.Append(ServicesVSResources.Unknown_parameters);
+                }
+                else
+                {
+                    FormatParameters(stringBuilder, method);
+                }
+
+                stringBuilder.Append(")");
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        private static bool FormatMethodName(StringBuilder stringBuilder, Type declaringType)
+        {
+            if (declaringType == null)
+            {
+                return false;
+            }
+
+            var isAsync = false;
+            var fullName = declaringType.FullName.Replace('+', '.');
+            if (typeof(IAsyncStateMachine).GetTypeInfo().IsAssignableFrom(declaringType))
+            {
+                isAsync = true;
+                stringBuilder.Append(AsyncMethodPrefix);
+                var start = fullName.LastIndexOf('<');
+                var end = fullName.LastIndexOf('>');
+                if (start >= 0 && end >= 0)
+                {
+                    stringBuilder.Append(fullName.Remove(start, 1).Substring(0, end - 1));
+                }
+                else
+                {
+                    stringBuilder.Append(fullName);
+                }
+            }
+            else
+            {
+                stringBuilder.Append(fullName);
+                stringBuilder.Append(".");
+            }
+
+            return isAsync;
+        }
+
+        private static void FormatGenericArguments(StringBuilder stringBuilder, Type[] genericTypeArguments)
+        {
+            if (genericTypeArguments.Length <= 0)
+            {
+                return;
+            }
+
+            stringBuilder.Append("[");
+            var firstTypeParam = true;
```

Comment:
simpler to just do things like:  stringBuilder.Append(string.Join(",", typeArgs.Select(g => g.Name)));